frontend/src/metabase/static-viz/lib/numbers.ts

```diff
@@ -24,53 +25,9 @@ const DEFAULT_OPTIONS = {
 };
 
 export const formatNumber = (number: number, options?: NumberFormatOptions) => {
-  const {
-    number_style,
-    currency,
-    currency_style,
-    number_separators: [decimal_separator, grouping_separator],
-    decimals,
-    scale,
-    prefix,
-    suffix,
-    compact,
-  } = handleSmallNumberFormat(number, { ...DEFAULT_OPTIONS, ...options });
+  const { prefix, suffix } = { ...DEFAULT_OPTIONS, ...options };
 
-  function createFormat(compact?: boolean) {
-    if (compact) {
-      return new Intl.NumberFormat("en", {
-        style: number_style !== "scientific" ? number_style : "decimal",
-        notation: "compact",
-        compactDisplay: "short",
-        currency: currency,
-        currencyDisplay: currency_style,
-        useGrouping: true,
-        maximumFractionDigits: decimals != null ? decimals : 2,
-      });
-    }
-
-    return new Intl.NumberFormat("en", {
-      style: number_style !== "scientific" ? number_style : "decimal",
-      notation: number_style !== "scientific" ? "standard" : "scientific",
-      currency: currency,
-      currencyDisplay: currency_style,
-      useGrouping: true,
-      minimumFractionDigits: decimals,
-      maximumFractionDigits: decimals != null ? decimals : 2,
-    });
-  }
-
-  const format = createFormat(compact);
-
-  const separatorMap = {
-    ",": grouping_separator || "",
-    ".": decimal_separator,
-  };
-  const formattedNumber = format
-    .format(number * scale)
-    .replace(/,|\./g, separator => separatorMap[separator as "." | ","]);
-
-  return `${prefix}${formattedNumber}${suffix}`;
+  return `${prefix}${appFormatNumber(number, options)}${suffix}`;
```

Comment:
unfortunately, it's hard to use `formatStaticValue` in some places since it expects a `column` to be included in the option.

https://github.com/metabase/metabase/blob/6cf42efe504a39d9e66cb8433fd45eb80fbe00cf/frontend/src/metabase/static-viz/lib/format.ts#L48

So a place like `<XYChart />` still needs to call `formatNumber` directly. In the future, we should use `formatStaticValue` instead.
https://github.com/metabase/metabase/blob/6cf42efe504a39d9e66cb8433fd45eb80fbe00cf/frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx#L200